app/Console/Commands/Migrations/FixContacts.php

```diff
@@ -0,0 +1,143 @@
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidInterface;
+use Spatie\Activitylog\Facades\CauserResolver;
+
+final class FixContacts extends Command
+{
+    use ReadCsvFile;
+
+    protected $signature = 'migrate:fix-contacts';
+
+    protected $description = 'Fix the missing technical and or functional contacts in the database.';
+
+    public function __construct(
+        private readonly ContactRepository $contactRepository,
+        private readonly InsightlyMappingRepository $insightlyMappingRepository
+    ) {
+        parent::__construct();
+    }
+
+    public function handle(): int
+    {
+        Model::unsetEventDispatcher();
+
+        CauserResolver::setCauser(UserModel::createSystemUser());
+
+        $this->info('Fixing contacts...');
+
+        // @phpstan-ignore-next-line
+        $integrationsIds = IntegrationModel::all()->pluck('id')->toArray();
+        $integrationsIdsCount = count($integrationsIds);
+
+        if ($integrationsIdsCount <= 0) {
+            $this->warn('No integrations to fix contacts for');
+            return 0;
+        }
+
+        if (!$this->confirm('Are you sure you want to fix ' . $integrationsIdsCount . ' integrations?')) {
+            return 0;
+        }
+
+        foreach ($integrationsIds as $integrationId) {
+            $this->info(PHP_EOL . '--- Start: Fixing contacts for integration ' . $integrationId);
+
+            $contacts = $this->contactRepository->getByIntegrationId(Uuid::fromString($integrationId));
+
+            $contributor = $contacts->first(fn (Contact $contact) => $contact->type === ContactType::Contributor);
+            $technical = $contacts->first(fn (Contact $contact) => $contact->type === ContactType::Technical);
+            $functional = $contacts->first(fn (Contact $contact) => $contact->type === ContactType::Functional);
+
+            if ($contributor === null) {
+                $this->warn('No contributor found for integration ' . $integrationId);
+                continue;
+            }
+
+            if ($technical !== null && $functional !== null) {
+                $this->info('Technical and functional contacts already exist for integration ' . $integrationId);
+                continue;
+            }
+
+            $this->info('Contributor found with id ' . $contributor->id . ' and email ' . $contributor->email);
+
+            try {
+                $contributorMapping = $this->insightlyMappingRepository->getByIdAndType(
+                    $contributor->id,
+                    ResourceType::Contact
+                );
+            } catch (ModelNotFoundException) {
+                $contributorMapping = null;
+                $this->warn('No contributor mapping found for integration ' . $integrationId);
+            }
+
+            $technical = $contacts->first(fn (Contact $contact) => $contact->type === ContactType::Technical);
```

Comment:
You already got the technical contact here no?

https://github.com/cultuurnet/publiq-platform/pull/1077/files#diff-eaca49e653648856b75c477c675a016d0ee2e145b6d6dab1c93e5fc4b2575f53R63-R65